### ISSUE TITLE
StringSpec test cases with multiline names broken

### ISSUE THREAD

**Issue by neelkamath** [kotest/kotest#1441]
# Bugs

I think I have found two bugs with `StringSpec`s. I'm unsure whether other testing styles are affected as well.

## Environment

Here's my environment. I'm unsure whether other environments are affected.

- Kotlin/JVM 1.3.72
- Kotest 4.0.5
- Gradle 6.4

## Bug 1: Test Failures

Consider this psuedocode.

```kotlin
class TestSuite : StringSpec({
    """
    test
    case
    1
    """ {
        // test
    }

    """
    test
    case
    2
    """ {
        // test
    }

    "test case 3" {
        // test 
    }
})
```

If I run the entire suite, and the first two test cases fail, then the error message used for one of the cases is used as the error message for the other case as well. Similarly, if only one of the first two cases fail, both fail with the error message of the failed case.

I _guess_ this happens even if there's more than one test case with a multiline name.

## Bug 2: Focusing

Focusing (prepending the test case's name with `f:`) doesn't work for test cases with multiline names. In essence, every test case runs even if I focus a test case with a multiline name.

I _guess_ the same holds true for bangs (prepending the test case's name with `!`).

**Comment by ashishkujoy** [comment]
@neelkamath For the first problem if you observe carefully the report mentions 3 test failure instead of 2, with two of them with same stacktrace(and failure message) but only one of them will have actual test name and other will ```executionError``` as test name. This is a known problem with reporting in kotest #1330 
<img width="1577" alt="Screenshot 2020-05-16 21 30 36" src="https://user-images.githubusercontent.com/34642693/82124409-82a94a80-97bc-11ea-828d-c5fab22fff23.png">

**Comment by ashishkujoy** [comment]
Two the second problem 
instead of this 
```kotlin
"""
f:test
case
2""" {

}
```
Use
```kotlin
"""f:
test
case
2"""
``` {
}

**Comment by sksamuel** [comment]
Ashish I think it's worth us calling .trim() on the names when we generate
the TestCase instances.

**Comment by neelkamath** [comment]
@ashishkujoy 

## Regarding Bug 1

Note: When I refer to "test case", I mean "test case with a multiline name".

I tried to reproduce bug 1 today with simpler code.

When one test case is supposed to pass, and another is supposed to fail, only one fails. This is not a bug.

However, bug 1 still exists, in the other scenario I mentioned. Consider the following code.

```kotlin
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class MyTest : StringSpec({
    """
    test
    case
    1
    """ { 1 shouldBe 2 }

    """
    test
    case 
    2
    """ { true shouldBe false }
})
```

Both tests are supposed to fail, and they both do. However, the test report shows test case 1's failure for test case 2's failure as well. Please unzip the attached folder and open `index.html` in your browser to see the Gradle report (I'm unsure whether this is a Gradle bug).

[Test Report.zip](https://github.com/kotest/kotest/files/4639514/Test.Report.zip)

## Regarding Bug 2

Your focusing technique works, and makes sense. It didn't occur to me that whitespace would've affected it, especially since Gradle replaces multiline test case names with single spaces. It might help to add this to the focus (and bang) docs.

**Comment by sksamuel** [comment]
This is a gradle limitation whereby it's taking only the first line from
the multi line, which is of course just a '\n' in this case, so they clash.
We should strip out new lines from multiline strings, as they won't work in
intellij or gradle. We already strip newlines in the intellij plugin.

**Comment by sksamuel** [comment]
Fix will be included in 4.0.6 and 4.1.0
